modules/node/src/utils.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  TransferWithExpiryStatus,
+  AppName,
+  TransferStatus,
+  TransferStatuses,
+  TransferWithExpiryStatuses,
+  HashLockTransferAppState,
+} from "@connext/types";
+import { AppInstance, AppType } from "./appInstance/appInstance.entity";
+import { bigNumberifyJson } from "@connext/utils";
+
+export function appStatusesToTransferStatus<T extends AppName>(
+  senderApp: AppInstance<T>,
+  receiverApp?: AppInstance<T>,
+): TransferStatus | undefined {
+  switch (senderApp.type) {
+    case AppType.PROPOSAL: {
+      return TransferStatuses.PENDING;
+    }
+    case AppType.UNINSTALLED: {
+      if (!receiverApp) {
+        // Sender app was uninstalled and receiver app rejected
+        return TransferStatuses.FAILED;
+      }
+      if (receiverApp.type !== AppType.UNINSTALLED) {
+        // receiver app exists, while sender app has been uninstalled
```

Comment:
This case is actually impossible to happen, we don't allow the sender to uninstall before the receiver. Maybe we should hard error here?